src/Http/Headers/src/MediaTypeHeaderValue.cs

```diff
@@ -76,7 +76,7 @@ public StringSegment Charset
         {
             get
             {
-                return NameValueHeaderValue.Find(_parameters, CharsetString)?.Value.Value;
+                return NameValueHeaderValue.Find(_parameters, CharsetString)?.Value ?? StringSegment.Empty;
             }
             set
             {
```

Comment:
Latest checks picked up build 20210827.29 from dotnet-runtime-official. That included the dotnet/runtime#58264 fix. Nonetheless, https://github.com/dotnet/aspnetcore/pull/35547/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.xml#diff-c3befa4d32f32a5ddad50d5be963e5522bb29330dac0b0f5ad4836864baaa8fdR118-R122 isn't validated as correct. Ideas @eerhardt @maxkoshevoi❔

Reply:
OIC: `IsNullOrEmpty` lacks `[MemberNotNullWhen(false, nameof(Value))]` (or `IsNullOrEmpty lacks [MemberNotNullWhen(false, nameof(Buffer))].`). That's a bug isn't it @eerhardt @maxkoshevoi❔

Reply:
Dang, you are right! I've completely missed `IsNullOrEmpty`. It lacks both attributes you described. And looking into it, there even more attributes missing. Created another PR:

https://github.com/dotnet/runtime/pull/58308

PS: There's also `IsNullOrEmpty` in `StringValues`, but there're no fields or properties to mark as non-null. Only `ToArray` and `this[0]`.